Bring the app content in line with the prototype

peteryates opened the pull request:
Prior to MVP we need to bring this app inline with the prototype.

Here's the list of things to fix from @Bethlt4 

Do now:

* [x] Fix the title on the homepage
* [x] Fix the content on the splash page
* [x] Do exact copy from prototype to app of the CCP index page (key stage content/rationale page)
* [x] Do exact copy from prototype to app of show CCP page (year 7 history curriculum)
* [x] Do exact copy from prototype to app of show unit page (Medieval realms unit)- keep two divisions of data: move benefits up
* [x] Number the tabs
* [x] Lesson plan- Battle of Hastings is Lesson 3 not 1 - we have 2 full lessons that can currently be added to the app
* [x] Do exact copy from prototype to app of knowledge overview tab - add the lesson structure guidance (rationale) 
* [x] Lesson contents tab (and all tabs), include text headers 
* [x] Downloads tab - remove the word ‘plan’ from the last sentence

-----

## Next sprint, after some planning...

* [ ]  **design task:** for changing radio buttons for year groups?
* [ ]  **dev task:** change rationale paragraphs in app to be divided into 2 sections that reflect supplier upload e.g. rationale, theory, planning, guidance (how to teach)?
* [ ] **separate task:** when a supplier doesn’t provide a piece of data e.g. ‘core knowledge’, needs to still be displayed but something like ‘not applicable’ to go underneath (might need a ‘not applicable’ option for certain fields in the supplier upload)



------

# Changes

| Prototype | Rails app |
| ----- | ----- |
| ![Screenshot from 2020-03-13 14-00-40](https://user-images.githubusercontent.com/128088/76627673-54e12480-6533-11ea-82e5-61d35e8599bc.png) | ![Screenshot from 2020-03-13 14-00-42](https://user-images.githubusercontent.com/128088/76627684-59a5d880-6533-11ea-853b-6e078399d2f6.png) |
| ![Screenshot from 2020-03-13 14-03-42](https://user-images.githubusercontent.com/128088/76627742-78a46a80-6533-11ea-8fad-94aba861a568.png) | ![Screenshot from 2020-03-13 14-03-44](https://user-images.githubusercontent.com/128088/76627760-7e01b500-6533-11ea-81c3-6bbde19d2d4e.png) |
| ![Screenshot from 2020-03-13 14-04-58](https://user-images.githubusercontent.com/128088/76627832-a8ec0900-6533-11ea-9925-692315d14750.png) | ![Screenshot from 2020-03-13 14-04-59](https://user-images.githubusercontent.com/128088/76627840-ac7f9000-6533-11ea-9648-bd59542268b2.png) || ![Screenshot from 2020-03-13 14-05-47](https://user-images.githubusercontent.com/128088/76627883-c325e700-6533-11ea-879c-6207011fd664.png) | ![Screenshot from 2020-03-13 14-05-49](https://user-images.githubusercontent.com/128088/76627894-c8833180-6533-11ea-8c60-69bca545dce6.png) |
| ![Screenshot from 2020-03-13 14-07-14](https://user-images.githubusercontent.com/128088/76628024-fa949380-6533-11ea-854f-447077186395.png) | ![Screenshot from 2020-03-13 14-07-16](https://user-images.githubusercontent.com/128088/76628029-fd8f8400-6533-11ea-9bd3-f6ce5e5ff675.png) |
| ![Screenshot from 2020-03-13 14-08-20](https://user-images.githubusercontent.com/128088/76628088-1c8e1600-6534-11ea-82d6-3e5e1fc6b82b.png) | ![Screenshot from 2020-03-13 14-08-22](https://user-images.githubusercontent.com/128088/76628091-1f890680-6534-11ea-8788-8d6d3132d015.png) |
| ![Screenshot from 2020-03-13 14-08-56](https://user-images.githubusercontent.com/128088/76628145-37608a80-6534-11ea-8f30-fa04d4c0d84b.png) | ![Screenshot from 2020-03-13 14-08-55](https://user-images.githubusercontent.com/128088/76628173-3cbdd500-6534-11ea-8c6f-0e819c1aa2d0.png) |
| ![Screenshot from 2020-03-13 14-10-26](https://user-images.githubusercontent.com/128088/76628344-7a226280-6534-11ea-9817-3a782b43aaf4.png) | ![Screenshot from 2020-03-13 14-10-28](https://user-images.githubusercontent.com/128088/76628357-7e4e8000-6534-11ea-9d2c-bd12e527a181.png) |

peteryates committed:
Make the homepage title less wordy

peteryates committed:
Update splash page copy

peteryates committed:
Add static CCP information to selection page

peteryates committed:
Add and configure YearsController

The years controller will break up the display of a CCP's units into
groups of units by school year

peteryates committed:
Fix capitalisation of Battle of Hastings

peteryates committed:
Add CompleteCurriculumProgramme#title

The title is comprised of the key stage and subject name

peteryates committed:
Move unit listing to YearsController

This makes more sense because a CCP contains units that span several
years but we need to display a list of units for a single year at once.

peteryates committed:
Update the CCP content with prototype copy

The seed data is now slightly incorrect (see the note in the medieval
realms file) but will make the application match the prototype for
testing

peteryates committed:
Remove card header link and make footer link bold

peteryates committed:
Fix ordering of lessons in a unit card

peteryates committed:
Rename the-norman-invasion to history/year-7

peteryates committed:
Rename the-battle-of-hastings to medieval-realms

peteryates committed:
Update medieval realms unit content from prototype

peteryates committed:
Add ordered_by_position scope to Unit

peteryates committed:
Order units by unit and lesson position

peteryates committed:
Make unit cards grey when there are no lessons

peteryates committed:
Add some placeholder units to Y7 history

peteryates committed:
Rename prelude-to-the-battle to battle-of-hastings

peteryates committed:
List the lesson summary in learning objectives

peteryates committed:
Ensure lessons ordered properly on units screen

peteryates committed:
Add some more spacing in the units table

peteryates committed:
Update lesson content to match prototype

peteryates committed:
Change 'Coming soon' to 'Available soon'

peteryates committed:
Number the tabs

This is an attempt to make users read the tabs sequentially, depsite the
tabs being sequential by the nature of their order.

There are several problems with this approach that may need to be
redressed:
 * '3 Downloads' makes it read like I have 'three things to download'
 * The tabs themselves are numbered using an ordered list, so screen
   readers *should* cope ok, but the tab buttons (at the bottom of each
   tab) aren't; I'm not sure how best to make it so the SR doesn't read
   the number out - or reads it out in a way that makes sense to the
   user that it'll take them to the next section

peteryates committed:
Remove the word 'plan' from instructions

peteryates committed:
Update lesson contents to match prototype

peteryates committed:
Make the lesson show screen match prototype

There are a couple of shortcuts here that are awaiting the update of
underlying tables with a more-recent set of attributes.

peteryates committed:
Fix failing specs

peteryates committed:
Fix breadcrumbs, they pointed to page that's moved

peteryates committed:
Fix tab list rendering on mobile

The design system formats this tab list with a long dash "\2014" when
narrower than tablet. As we're unconventionally using a ol rather than
ul, remove the left padding the dash appears *after* the number but
before the list item content.

rjlynch commented on app/views/teachers/units/show.slim:
```diff
@@ -5,17 +5,15 @@
 
   .govuk-grid-column-one-third
-    = render 'siblings', units: @unit.complete_curriculum_programme.units
+    = render 'siblings', units: @unit.complete_curriculum_programme.units.ordered_by_position
```
We should probably disable the links to other units if they have no content (like on the year show view)

peteryates replied:
Yeah, well spotted :owl:

rjlynch commented on app/webpacker/styles/_lessons.scss:
```diff
@@ -1,10 +1,42 @@
+      // the design system formats this tab list with a long dash "\2014" when narrower than
+      // tablet. As we're unconventionally using a ol rather than ul, remove the left padding
+      // the dash appears *after* the number but before the list item content
+      &::before {
```
Nice

rjlynch commented on app/webpacker/styles/_units.scss:
```diff
@@ -1,21 +1,30 @@
 // units table
```
Not sure where to add this but it's todo with the unit page.
When viewing the unit#show page on a small screen the side bar breaks above the lesson table which seems a bit odd
![Screenshot 2020-03-13 at 16 42 38](https://user-images.githubusercontent.com/9936028/76641738-fb382480-6549-11ea-802a-e1585dd13327.png)

peteryates replied:
Ah yeah, I tried to make the table wider by placing it in a separate row underneath. I'll move it up to the same column as the text content and it should flow in the right order :+1:

rjlynch reviewed (approved):
LGTM some minor comments.
I'm going to have to self isolate to get the merge conflicts in db/seeds resolved when my feature branch is done! 😿

rjlynch commented:
Heroku will need reseeding once this is merged

peteryates committed:
Fix ordering of units page on mobile

I'd moved the table to its own row so it could take up the full width of
the page but this affects the ordering of content when viewed on mobile.

This can be overcome with media queries but for the time being just
setting it back to how it was previously

Refs #137

peteryates committed:
Don't link to units without any lessons

While we're in MVP we have some units that have no lessons. These might
confuse users so for the time being stop linking to them. Comments are
left in to remind us to remove this functionality when ready

Refs #137
